File: leetcode/Algorithms/389.FindTheDifference/Solution.java
public class Solution {
    public char findTheDifference(String s, String t) {
        int cache[] = new int[26];

        for (char sc: s.toCharArray()) {
            cache[sc - 'a']++;
        }

        for (char tc: t.toCharArray()) {
            if (cache[tc - 'a'] > 0) {
                cache[tc - 'a']--;
            } else {
                return tc;
            }
        }

        return t.charAt(t.length() - 1);
    }
}

File: leetcode/Algorithms/389.FindTheDifference/Solution.c
char findTheDifference(char* s, char* t) {
    int result = 0;
    
    for (int i = 0; s[i] != 0; i++) {
        result ^= s[i];
    }
    
    for (int j = 0; t[j] != 0; j++) {
        result ^= t[j];
    }

    return (char) result;
}
